astropy/io/tests/test_registry_help.py

```diff
@@ -138,3 +147,48 @@ def test_ccddata_read_help_fits():
     assert "CCDData.read(format='fits') documentation" in doc
     assert "Generate a CCDData object from a FITS file" in doc
     assert "hdu_uncertainty : str or None, optional" in doc
+
+
+def test_table_write_help_jsviewer():
+    """
+    Test dynamically created documentation help via the I/O registry for
+    'jsviewer'.
+    """
+    out = StringIO()
+    Table.write.help('jsviewer', out)
+    doc = out.getvalue()
+
+    # Check a smattering of expected content
+    assert "Table.write general documentation" not in doc
+    assert "The available built-in formats" not in doc
+    assert "Table.write(format='jsviewer') documentation" in doc
+
+
+def test_table_read_no_help():
+    try:
+        io_registry.register_reader('test1', Table, empty_reader)
```

Comment:
No sure what's the best way to register temporarily a reader/writer. I saw in other tests that another class was used (`TestData`) instead of `Table`, but `TestData.read.help` does not work...

Reply:
I'd be OK on this with just the one test that demonstrates fixing the problem with jsviewer highlighted in the description.  I'd have to review the code and prior examples, but honestly this seems like too much effort for this small fix which is clearly correct.

Reply:
Yes agreed :smile:, but if I or someone else update the jsviewer docstring in the future it may lead to an inconsistent test/docstring, so I didn't want to rely on the jsviewer test for this. As you prefer, I'm happy to remove the two extra tests if using the registry like this is overkill.

Reply:
You are right that the test could in the future be a problem, but it is not highly likely. I have come to accept the reality of limited resources and that "perfect is the enemy of good". So I recommend to just get this in with an OK test and move on to the next one out of hundreds of issues to deal with. If it breaks later we'll deal.

Reply:
Ok, I removed the tests and added the changelog entry. And squashed and rebased to get the fix for the docs build.